src/transaction.ts

```diff
@@ -103,128 +126,108 @@ export const sendTransaction = async function (
-        signer: t.signer as TransactionSignerAccount,
-        id: i,
-      })),
-      (t) => t.signer.addr,
-    )
-
-    // Perform the signature, grouped by signer
-    const signed = (
-      await Promise.all(
-        Object.values(groupedBySigner).map(async (signature) => {
-          const indexes = signature.map((s) => s.id)
-          const signed = await signature[0].signer.signer(group, indexes)
-          return signed.map((txn, i) => ({
-            txn,
-            id: indexes[i],
-          }))
-        }),
-      )
-    ).flatMap((a) => a)
-
-    // Extract the signed transactions in order
-    signedTransactions = signed.sort((s1, s2) => s1.id - s2.id).map((s) => s.txn)
+  let groupId: string | undefined = undefined
+  if (transactionsToSend.length === 1) {
+    Config.getLogger(sendParams?.suppressLog).info(`Sending transaction ${transactionsToSend[0].txID()}`)
   } else {
-    signedTransactions = group.map((groupedTransaction, id) => {
-      const signer = transactionsWithSigner[id].signer
-      return 'sk' in signer
-        ? groupedTransaction.signTxn(signer.sk)
-        : 'lsig' in signer
-        ? algosdk.signLogicSigTransactionObject(groupedTransaction, signer).blob
-        : 'sign' in signer
-        ? signer.sign(groupedTransaction)
-        : // This bit will never happen because above if statement
-          new Uint8Array()
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    groupId = transactionsToSend[0].group ? Buffer.from(transactionsToSend[0].group).toString('base64') : ''
+    Config.getLogger(sendParams?.suppressLog).info(`Sending group of ${transactionsToSend.length} transactions (${groupId})`, {
+      transactionsToSend,
     })
-  }
-
-  Config.getLogger(sendParams?.suppressLog).debug(
-    `Signer IDs (${groupId})`,
-    transactionsWithSigner.map((t) => getSenderAddress(t.signer)),
-  )
 
-  Config.getLogger(sendParams?.suppressLog).debug(
-    `Transaction IDs (${groupId})`,
-    transactionsToSend.map((t) => t.txID()),
-  )
+    Config.getLogger(sendParams?.suppressLog).debug(
+      `Transaction IDs (${groupId})`,
+      transactionsToSend.map((t) => t.txID()),
+    )
+  }
 
-  // https://developer.algorand.org/docs/rest-apis/algod/v2/#post-v2transactions
-  await algod.sendRawTransaction(signedTransactions).do()
+  const result = await atc.execute(algod, sendParams?.maxRoundsToWaitForConfirmation ?? 5)
 
   Config.getLogger(sendParams?.suppressLog).info(`Group transaction (${groupId}) sent with ${transactionsToSend.length} transactions`)
 
   let confirmations: PendingTransactionResponse[] | undefined = undefined
   if (!sendParams?.skipWaiting) {
     confirmations = await Promise.all(
-      transactionsToSend.map(async (t) => await waitForConfirmation(t.txID(), sendParams?.maxRoundsToWaitForConfirmation ?? 5, algod)),
+      transactionsToSend.map(async (t) => (await algod.pendingTransactionInformation(t.txID()).do()) as PendingTransactionResponse),
```

Comment:
Unfortunately, ATC doesn't expose the PendingTransactionInfo so I need to make duplicate calls to algod :(